sspl_test/alerts/realstor/test_real_stor_dg_sensor.py

```diff
@@ -0,0 +1,141 @@
+            time.sleep(0.1)
+            print(exception)
+
+    assert(dg_sensor_msg is not None)
+    assert(dg_sensor_msg.get("alert_type") is not None)
+    assert(dg_sensor_msg.get("alert_id") is not None)
+    assert(dg_sensor_msg.get("severity") is not None)
+    assert(dg_sensor_msg.get("host_id") is not None)
+    assert(dg_sensor_msg.get("info") is not None)
+
+    dg_sensor_info = dg_sensor_msg.get("info")
+    assert(dg_sensor_info.get("resource_type") is not None)
+    assert(dg_sensor_info.get("event_time") is not None)
+    assert(dg_sensor_info.get("resource_id") is not None)
+
+    dg_sensor_specific_info = dg_sensor_msg.get("specific_info")
+    assert(dg_sensor_specific_info is not None)
+    assert(dg_sensor_specific_info.get("cvg_name") == 'A01')
+    assert(dg_sensor_specific_info.get("cvg_id") == 0)
+    assert(dg_sensor_specific_info.get("object_name") is not None)
+    assert(dg_sensor_specific_info.get("name") is not None)
+    assert(dg_sensor_specific_info.get("size") is not None)
+    assert(dg_sensor_specific_info.get("freespace") is not None)
+    assert(dg_sensor_specific_info.get("storage_type") is not None)
+    assert(dg_sensor_specific_info.get("pool") is not None)
+    assert(dg_sensor_specific_info.get("pool_serial_number") is not None)
+    assert(dg_sensor_specific_info.get("pool_percentage") is not None)
+    assert(dg_sensor_specific_info.get("owner") is not None)
+    assert(dg_sensor_specific_info.get("raidtype") is not None)
+    assert(dg_sensor_specific_info.get("status") is not None)
+    assert(dg_sensor_specific_info.get("create_date") is not None)
+    assert(dg_sensor_specific_info.get("disk_description") is not None)
+    assert(dg_sensor_specific_info.get("serial_number") is not None)
+    assert(dg_sensor_specific_info.get("pool_sector_format") is not None)
+    assert(dg_sensor_specific_info.get("health") is not None)
+    assert(dg_sensor_specific_info.get("health_reason") is not None)
+    assert(dg_sensor_specific_info.get("health_recommendation") is not None)
+
+def check_sspl_ll_is_running():
+    # Check that the state for sspl_ll service is active
+    found = False
+
+    # Support for python-psutil < 2.1.3
+    for proc in psutil.process_iter():
+        if proc.name == "sspl_ll_d" and \
+           proc.status in (psutil.STATUS_RUNNING, psutil.STATUS_SLEEPING):
+               found = True
+
+    # Support for python-psutil 2.1.3+
+    if found == False:
+        for proc in psutil.process_iter():
+            pinfo = proc.as_dict(attrs=['cmdline', 'status'])
+            if "sspl_ll_d" in str(pinfo['cmdline']) and \
+                pinfo['status'] in (psutil.STATUS_RUNNING, psutil.STATUS_SLEEPING):
+                    found = True
+
+    assert found == True
+
+    # Clear the message queue buffer out
+    while not world.sspl_modules[IngressProcessorTests.name()]._is_my_msgQ_empty():
+        world.sspl_modules[IngressProcessorTests.name()]._read_my_msgQ()
```

Comment:
this function is also present in `sspl_test/common.py`

Reply:
Done